packages/cryptoplease_link/jupiter_aggregator/lib/src/model.dart

```diff
@@ -150,15 +138,50 @@ class QuoteRequestDto with _$QuoteRequestDto {
 }
 
 @freezed
-class SwapRequestDto with _$SwapRequestDto {
-  const factory SwapRequestDto({
+class JupiterSwapRequestDto with _$JupiterSwapRequestDto {
+  const factory JupiterSwapRequestDto({
     required String userPublicKey,
     required JupiterRoute route,
     bool? wrapUnwrapSOL,
     String? feeAccount,
     String? destinationWallet,
-  }) = _SwapRequestDto;
+  }) = _JupiterSwapRequestDto;
+
+  factory JupiterSwapRequestDto.fromJson(Map<String, dynamic> json) =>
+      _$JupiterSwapRequestDtoFromJson(json);
+}
+
+@freezed
+class JupiterSwapResponseDto with _$JupiterSwapResponseDto {
+  const factory JupiterSwapResponseDto({
+    required String? setupTransaction,
+    required String swapTransaction,
+    required String? cleanupTransaction,
+  }) = _JupiterSwapResponseDto;
+
+  factory JupiterSwapResponseDto.fromJson(Map<String, dynamic> json) =>
+      _$JupiterSwapResponseDtoFromJson(json);
+}
+
+@freezed
+class PriceRequestDto with _$PriceRequestDto {
+  const factory PriceRequestDto({
+    required String id,
+  }) = _PriceRequestDto;
+
+  factory PriceRequestDto.fromJson(Map<String, dynamic> json) =>
+      _$PriceRequestDtoFromJson(json);
+}
 
-  factory SwapRequestDto.fromJson(Map<String, dynamic> json) =>
-      _$SwapRequestDtoFromJson(json);
+@freezed
+class PriceResponseDto with _$PriceResponseDto {
+  const factory PriceResponseDto({
+    @JsonKey(name: 'data', readValue: _readPrice) required double price,
```

Comment:
This is very confusing. Make it closer to the real structure, i.e. add `Data(double price)` intermediate DTO.